src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -260,6 +260,31 @@ public void SslStream_StreamToStream_Write_ReadByte_Success()
 
         [OuterLoop] // TODO: Issue #11345
         [Fact]
+        public async Task SslStream_StreamToStream_WriteAsync_ReadByte_Success()
+        {
+            VirtualNetwork network = new VirtualNetwork();
```

Comment:
Would be interesting to have a test for the specific case that prompted this change - a task blocked on `ReadAsync` waiting for that, while the test calls `WriteAsync`. The inner stream should be guaranteed to not implement begin/end methods (what if `VirtualNetworkStream` is changed in the future?).